win32/devdoc/job_object_helper.md

```diff
@@ -30,6 +33,65 @@ MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_create);
 **SRS_JOB_OBJECT_HELPER_18_019: [** `job_object_helper_create` shall succeed and return the `JOB_OBJECT_HELPER` object.  **]**
 
 
+## job_object_helper_create_with_name
+```c
+MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_create_with_name, const char*, job_name);
+```
+``job_object_helper_create_with_name`` is used to create a Job Object with the specified name.
+
+**SRS_JOB_OBJECT_HELPER_19_001: [** `job_object_helper_create_with_name` shall log the error and return `NULL` if the passed `job_name` is `NULL` or empty. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_002: [** `job_object_helper_create_with_name` shall allocate `JOB_OBJECT_HELPER` object. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_016: [** `job_object_helper_create_with_name` shall call `mbs_to_wcs` to convert job_name to `wchar_t*`. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_003: [** `job_object_helper_create_with_name` shall call `CreateJobObject` to create a new job object passing `NULL` for `lpJobAttributes` and `job_name` to `lpName`. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_004: [** `job_object_helper_create_with_name` shall log the error and return `NULL` if the `CreateJobObject` fail. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_005: [** `job_object_helper_create_with_name` shall return the `JOB_OBJECT_HELPER` object when the job object is created successfully. **]**
+
+
+## job_object_helper_get
+```c
+MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_get, const char*, job_name);
+```
+``job_object_helper_get`` is used for getting a `JOB_OBJECT_HELPER` created with the name `job_name`.
+
+**SRS_JOB_OBJECT_HELPER_19_006: [** `job_object_helper_get` shall log the error and return `NULL` if the passed `job_name` is `NULL` or empty. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_017: [** `job_object_helper_get` shall call `mbs_to_wcs` to convert the `job_name` to `wchar_t*` type. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_007: [** `job_object_helper_get` shall call `OpenJobObject` with name parameter set to `job_name`. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_008: [** `job_object_helper_get` shall log the error and return `NULL` if the `job object` with the name does not exist. **]**
+
+**SRS_JOB_OBJECT_HELPER_19_009: [** `job_object_helper_get`shall return the `JOB_OBJECT_HELPER` object when the job object is opened successfully. **]**

```

Comment:
>`JOB_OBJECT_HELPER` object when the job object is opened successfully. [](http://example.com/codeflow?start=76&length=70)

This is very weird and I don't think it can work.

So assume 2 calls are previously done:
1) `job_object_helper_create_with_name("IONESCU")`
2) `job_object_helper_create_with_name("POPESCU")`

Now a call to `job_object_helper_get` is made with "IONESCU". 

Windows will return the `HANDLE` for IONESCU, but "our" object (THANDLE(JOB_OBJECT_HELPER))... how is it retrieved? How does the software know how to look for HANDLE(of IONESCU) => THANDLE(JOB_OBJECT_HELPER) in the absence of some global data... tracking all the strings<===>THANDLE(JOB_OBJECT_HELPER)?

==

So either I don't understand something basic, or there's missing requirements, or it cannot work.

Reply:
Probably: more explanations are required.

Reply:
This is reason to be "Waiting for author" for now

Reply:
Probably "will allocate" from your comment sounds like a requirement. (code needs to do some malloc somewhere)